#7 Maya v1.0.0 Simple Setup (#32)
Inbibo/adonisfx_docs: docs/documentation/maya/simple_setup.md

# A Simple Setup

> This page is dedicated to explain, step by step, a simple process of creating and setting every Adonis deformer in Maya. The scenarios presented here will are intended to provide the minimum required configurations to obtain plausible results.

Reviewer: ```suggestion
This page is dedicated to explain, step by step, a simple process of creating and setting every Adonis deformer in Maya. The scenarios presented here are intended to provide the minimum required configurations to obtain plausible results.
```

## AdnSkin Simple Setup

> To create a basic scenario using the AdnSkin deformer start with a scene with the following elements:

Reviewer: Done.

  - A target mesh with animation.
  - A simulated mesh (without animation).

The AdnSkin deformer will get applied to the simulated mesh.

<figure>
  <img src="images/setup_skin_0.png"> 

>   <figcaption>Figure 1: Basic setup for skin simulations.</figcaption>

Reviewer: Add bold style to all "Figure x:
```suggestion
  <figcaption><b>Figure 1</b>: Basic setup for skin simulations.</figcaption>
```

Reviewer: Done.

</figure>

### 1. Creating the deformer

To create the AdnSkin deformer select the target mesh and then the skin mesh (simulated mesh). Then press the ![AdnMuscle](images/adn_skin.png) shelf button or go to AdonisFX Menu > *Skin*.

> To create the AdnSkin deformer with some initial customization, double click the shelf button or press the option box in the menu item. This will display a pop-up window that will allow to do some initial customization, as well as creating the deformer with a custom name. Once all data has been provided press the "Create" button and the deformer will get created.

Reviewer: To keep consistency:
```suggestion
To create the AdnSkin deformer with some initial customization, double-click the shelf button or press the option box in the menu item. This will display a pop-up window that will allow to do some initial customization, as well as creating the deformer with a custom name. Once all data has been provided press the *Create* button and the deformer will get created.
```

Reviewer: Done.

<figure>
  <img src="images/setup_skin_1.png"> 
  <figcaption>Figure 2: AdnSkin deformer creation scenario.</figcaption>
</figure>

### 2. Painting weights

> Once the skin deformer is properly created it is possible now to paint its weights to correctly setup the deformer properties. To do so, select the simulated mesh and press the ![paint tool](images/adn_paint_tool.png) shelf button or go to AdonisFX Menu > *Paint Tool*.

Reviewer: Done.

</figure>

Finally, select the *Slide Constraints* attribute and paint weights only in those areas where the skin is supposed to slide over the reference mesh. In this case, focus these weights over the scapulas and articulations, as they are the areas of the skin with the most noticeable sliding.

<figure>
  <img src="images/setup_skin_4.png"> 
  <figcaption>Figure 5: Slide Constraints weights paint.</figcaption>
</figure>

With this basic paint setup the AdnSkin deformer will already show plausible results, expected of the skin to the reference target mesh. However, the possible parameters and tweaks to display high fidelity dynamics can be seen in the documentation for [AdnSkin](skin.md).

## AdnMuscle Simple Setup

To create a basic scenario using the AdnMuscle deformer, start with a scene with the following elements:

 - An animated rig.
 - A simulated mesh.

In this case the proposed example is to simulate a biceps in an animated full body rig. The AdnMuscle deformer will be applied to the mesh of the biceps.

<figure>
  <img src="images/setup_muscle_0.png"> 
  <figcaption>Figure 6: Basic setup for biceps simulations.</figcaption>
</figure>

### 1. Creating the deformer

> To create the AdnMuscle deformer, select the two joints of the rig that form the "arm" and the "forearm" and then the mesh of the muscle. Then, press the ![AdnMuscle](images/adn_muscle.png) shelf button or go to AdonisFX Menu > *Muscle*.

Reviewer: Done.

<figure>
  <img src="images/setup_muscle_1.png"> 
  <figcaption>Figure 7: AdnMuscle deformer creation scenario.</figcaption>
</figure>

> To create the AdnMuscle deformer with some initial customization, double click the shelf button or press the option box in the menu item. This will display a pop-up window that will allow to do some initial customization, as well as creating the deformer with a custom name. Once all data has been provided press the "Create" button and the deformer will get created.

Reviewer: To keep consistency:
```suggestion
To create the AdnMuscle deformer with some initial customization, double-click the shelf button or press the option box in the menu item. This will display a pop-up window that will allow to do some initial customization, as well as creating the deformer with a custom name. Once all data has been provided press the *Create* button and the deformer will get created.
```

Reviewer: Done.

<figure>
  <img src="images/setup_muscle_2.png"> 
  <figcaption>Figure 8: AdnMuscle custom creation UI.</figcaption>
</figure>

### 2. Painting weights

> Once the muscle deformer is properly created it is possible now to paint its weights to correctly setup the deformer properties. To do so, select the simulated mesh and press the ![paint tool](images/adn_paint_tool.png) shelf button or go to AdonisFX Menu > *Paint Tool*.

Reviewer: ```suggestion
Once the AdnMuscle deformer is properly created it is possible now to paint its weights to correctly setup the deformer properties. To do so, select the simulated mesh and press the ![paint tool](images/adn_paint_tool.png) shelf button or go to AdonisFX Menu > *Paint Tool*.
```

Reviewer: Done.

<figure>
  <img src="images/setup_muscle_3.png"> 
  <figcaption>Figure 9: Attachment influence for joint 1.</figcaption>
</figure>

Start by painting attachment weights, painting the influence for each attachment by selecting the corresponding attachment from the list and painting its desired influence.  

<figure>
  <img src="images/setup_muscle_4.png"> 
  <figcaption>Figure 10: Attachment influence for joint 2.</figcaption>
</figure>

> After that is done, paint the muscle tendon weights, by selecting the "Tendon" attribute from the "Attribute" enumerator and paint over the parts of the muscle that should have tendon tissue.

Reviewer: ```suggestion
After that is done, paint the muscle tendon weights, by selecting the *Tendon* attribute from the *Attribute* enumerator and paint over the parts of the muscle that should have tendon tissue.
```

<figure>
  <img src="images/setup_muscle_5.png"> 
  <figcaption>Figure 12: Tendon weights for biceps.</figcaption>
</figure>

> Once tendons are painted, when selecting the "Fibers" attribute from the "Attribute" enumerator, painted fibers will be displayed, with a default direction set by the painted tendons. It is now possible to freely comb these fibers if it is desired.

Reviewer: ```suggestion
Once tendons are painted, when selecting the *Fibers* attribute from the *Attribute* enumerator, painted fibers will be displayed, with a default direction set by the painted tendons. It is now possible to freely comb these fibers if it is desired.
```

Reviewer: Done.

To change the fiber size or its color, go to the Attribute Editor, in the debug submenu, and customize the color, width and length of the drawn lines.

<figure>
  <img src="images/setup_muscle_6.png"> 
  <figcaption>Figure 13: Muscle fibers combing.</figcaption>
</figure>

Pressing play should display the muscle properly attached and with the expected dynamics. However, it still won't respond to external inputs like, for example, flexing an arm which add rigidity to the muscle.

### 3. Connecting AdnSensors

To have the muscle change and responding to external inputs (in this case an arm flex), it is recommended to setup and AdnSensor and connect it to the deformer. 

> To do this, first create a locator and a sensor. Create a rotation locator and sensor to read the elbow angle. Both elements can be created by selecting the three joints from which to create the rotation locator and sensor (arm, forearm and wrist joints) and directly click on the ![adnRotationSensor](images/adn_angle_sensor.png) shelf button or go to AdonisFX Menu > Sensors (on the "Create" group) > *Rotation*. With this, both a locator and its corresponding sensor will get created at the same time.

Reviewer: ```suggestion
To do this, first create a rotation locator and sensor to compute the elbow angle. Both elements can be created by selecting the three joints from which to create the rotation locator and sensor (arm, forearm and wrist joints) and directly click on the ![adnRotationSensor](images/adn_angle_sensor.png) shelf button or go to AdonisFX Menu > Sensors (on the *Create* group) > *Rotation*. With this, both a locator and its corresponding sensor will get created at the same time.
```

Reviewer: Done.

<figure>
  <img src="images/setup_muscle_7.png"> 
  <figcaption>Figure 14: Rotation locator and sensor setup in elbow.</figcaption>
</figure>

> Now that the sensor is created it has to be connected to the deformer. To do so, make use of the Connection editor, Which must be opened from the AdonisFX Menu > Sensors (on the "Edit" group) > *Connection Editor*.

Reviewer: ```suggestion
Now that the sensor is created it has to be connected to the deformer. To do so, make use of the Connection editor, which must be opened from the AdonisFX Menu > Sensors (on the *Edit* group) > *Connection Editor*.
```

Reviewer: Done.

> Once the Connection Editor is open, select the locator from the scene and press the *Reload Left* button, then select the simulated mesh and press the *Reload Right* button. Once this is done, on the lists (below the buttons) the selections done on each side with their respective connectable attributes will appear. Connect the "activationAngle" attribute from the locator to the "activation" attribute from the deformer.

Reviewer: ```suggestion
Once the Connection Editor is open, select the locator from the scene and press the *Reload Left* button, then select the simulated mesh and press the *Reload Right* button. Once this is done, on the lists (below the buttons) the selections done on each side with their respective connectable attributes will appear. Connect the *activationAngle* attribute from the locator to the *activation* attribute from the deformer.
```

Reviewer: Done.

<figure>
  <img src="images/setup_muscle_8.png"> 
  <figcaption>Figure 15: Connection Editor tool.</figcaption>
</figure>

When the elbow is flexed (and therefore the angle from the locator gets smaller) the muscle activation will get higher, simulating a much more realistic scenario.

To tweak additional parameters of the AdnMuscle deformer, the following documentation for [AdnMuscle](muscle.md) can be consulted.

## AdnRibbonMuscle Simple Setup

The process to setup an AdnRibbonMuscle is very similar to the one of setting up and AdnMuscle. It essentially follows the same steps. Start with the following elements:

 - An animated rig.
 - A simulated mesh.

In this case a planar muscle will be simulated corresponding to a biceps, which will yield similar results to the case of the AdnMuscle deformer previously shown.

<figure>
  <img src="images/setup_ribbon_muscle_0.png"> 
  <figcaption>Figure 16: Basic setup for planar biceps simulations.</figcaption>
</figure>

### 1. Creating the deformer

> Similar to AdnMuscle, create the deformer by selecting the attachments (arm and forearm joints) and then the mesh to deform (the biceps muscle) and then pressing the ![AdnMuscle](images/adn_ribbon_muscle.png) shelf button or go to AdonisFX Menu > *Ribbon Muscle*.

Reviewer: ```suggestion
Similar to AdnMuscle, create the AdnRibbonMuscle deformer by selecting the attachments (arm and forearm joints) and then the mesh to deform (the biceps muscle) and then pressing the ![AdnMuscle](images/adn_ribbon_muscle.png) shelf button or go to AdonisFX Menu > *Ribbon Muscle*.
```

Reviewer: Done.

<figure>
  <img src="images/setup_ribbon_muscle_1.png"> 
  <figcaption>Figure 17: AdnRibbonMuscle deformer creation scenario.</figcaption>
</figure>

> To create the AdnRibbonMuscle deformer with some initial customization, double click the shelf button or press the option box in the menu item. This will display a pop-up window that will allow to do some initial customization, as well as creating the deformer with a custom name. Once all data has been provided press the "Create" button and the deformer will get created.

Reviewer: ```suggestion
To create the AdnRibbonMuscle deformer with some initial customization, double-click the shelf button or press the option box in the menu item. This will display a pop-up window that will allow to do some initial customization, as well as creating the deformer with a custom name. Once all data has been provided press the *Create* button and the deformer will get created.
```

Reviewer: Done.

<figure>
  <img src="images/setup_ribbon_muscle_2.png"> 
  <figcaption>Figure 18: AdnRibbonMuscle custom creation UI.</figcaption>
</figure>

### 2. Painting weights

Once the muscle deformer is properly created it is possible now to paint its weights to correctly setup the deformer properties. To do so, select the simulated mesh and press the ![paint tool](images/adn_paint_tool.png) shelf button or go to AdonisFX Menu > *Paint Tool*.

Start by painting attachment weights, painting the influence for each attachment by selecting the corresponding attachment from the list and painting its desired influence.  

<figure>
  <img src="images/setup_ribbon_muscle_4.png"> 
  <figcaption>Figure 20: Attachment influence for joint 1.</figcaption>
</figure>

<figure>
  <img src="images/setup_ribbon_muscle_5.png"> 
  <figcaption>Figure 21: Attachment influence for joint 2.</figcaption>
</figure>

> After that is done, paint the muscle tendon weights, by selecting the "Tendon" attribute from the "Attribute" enumerator and painting over the parts of the muscle to represent tendon tissue.

Reviewer: ```suggestion
After that is done, paint the muscle tendon weights, by selecting the *Tendon* attribute from the *Attribute* enumerator and painting over the parts of the muscle to represent tendon tissue.
```

<figure>
  <img src="images/setup_ribbon_muscle_6.png"> 
  <figcaption>Figure 22: Tendon weights for planar biceps.</figcaption>
</figure>

> Now that tendons are painted, when selecting the "Fibers" attribute from the "Attribute" enumerator, painted fibers will be displayed, with a default direction set by the painted tendons. It is now possible to freely comb these fibers if it is desired.

Reviewer: ```suggestion
Now that tendons are painted, when selecting the *Fibers* attribute from the *Attribute* enumerator, painted fibers will be displayed, with a default direction set by the painted tendons. It is now possible to freely comb these fibers if it is desired.
```

Reviewer: Done.

In case the fiber or its color has to be manipulated, go to the Attribute Editor, in the debug submenu, and customize the color, width and length of the drawn lines.

<figure>
  <img src="images/setup_ribbon_muscle_7.png"> 
  <figcaption>Figure 23 : Muscle fibers combing.</figcaption>
</figure>

> Pressing play should show the muscle properly attached and with the expected dynamics. However, it still won't respond to external inputs like, for example, flexing an arm which add rigidity to the muscle.

Reviewer: Avoid contractions.
```suggestion
Pressing play should show the muscle properly attached and with the expected dynamics. However, it still will not respond to external inputs like, for example, flexing an arm which add rigidity to the muscle.
```

Reviewer: Done.

### 3. Connecting AdnSensors

The process to connect and AdnSensor to an AdnRibbonMuscle is the exact same to the one followed [for AdnMuscle](#3-connecting-adnsensors).

<figure>
  <img src="images/setup_ribbon_muscle_8.png"> 
  <figcaption>Figure 24: Connection Editor tool with AdnRotation sensor connected to AdnRibbonMuscle.</figcaption>
</figure>

> To tweak additional parameters of the AdnMuscle deformer, the following documentation for [AdnRibbonMuscle](ribbon.md) can be consulted.

Reviewer: ```suggestion
To tweak additional parameters of the AdnRibbonMuscle deformer, the following documentation for [AdnRibbonMuscle](ribbon.md) can be consulted.
```

Reviewer: Done.

## AdnSimshape

To create a basic scenario using the AdnSimshape deformer, start with a scene with the following elements:

 - An animated facial mesh (to which to apply the deformer).
 - A rest mesh.
 - A deformation mesh with only the facial animation.

All these meshes must have the same number of vertices and correspond to the same facial model.

<figure>
  <img src="images/setup_simshape_0.png"> 
  <figcaption>Figure 25: Basic setup for facial simulations.</figcaption>
</figure>

### 1. Creating the deformer

To create the AdnSimshape deformer it is required to select first the rest mesh and then the animated mesh. In this scenario, the animated mesh will be used as the simulated mesh.

Press the ![AdnMuscle](images/adn_simshape.png) shelf button or go to AdonisFX Menu > *Simshape*.

> To create the AdnSimshape deformer with some initial customization double click the shelf button or press the option box in the menu item. This will display a pop-up window that will allow to do some initial customization, as well as creating the deformer with a custom name. Once all data has been provided press the "Create" button and the deformer will get created.

Reviewer: ```suggestion
To create the AdnSimshape deformer with some initial customization double-click the shelf button or press the option box in the menu item. This will display a pop-up window that will allow to do some initial customization, as well as creating the deformer with a custom name. Once all data has been provided press the *Create* button and the deformer will get created.
```

Reviewer: Done.

<figure>
  <img src="images/setup_simshape_1.png"> 
  <figcaption>Figure 26: AdnSimshape deformer creation scenario.</figcaption>
</figure>

To add the deformation mesh to the deformer first select the deformation mesh, then the simulated mesh (it is the animation mesh), and then go to AdonisFX Menu > Simshape (edit group) > Add *Deform Mesh*. A message will notify is that the addition of the rest mesh has been done correctly.

<figure>
  <img src="images/setup_simshape_2.png"> 
  <figcaption>Figure 27: Addition of deform mesh to AdnSimshape deformer.</figcaption>
</figure>

### 2. Painting weights

> In the case of the AdnSimshape use the Maya Paint tool to setup and paint its paintable weight attributes. In this section painting the Attraction Force weights will be exposed, as this is the value that dictates how much of each simulated vertex should follow the animation. This value is flooded by default to 1.0, meaning that by default the simulated mesh will follow completely the animation, without displaying dynamics.

Reviewer: ```suggestion
In the case of the AdnSimshape use the Maya Paint tool to setup and paint its paintable weight attributes. In this section painting the *Attraction Force* weights will be exposed, as this is the value that dictates how much of each simulated vertex should follow the animation. This value is flooded by default to 1.0, meaning that by default the simulated mesh will follow completely the animation, without displaying dynamics.
```

Reviewer: Done.

> In high deformation areas, such as around the mouth or under the eyes, add medium to low values (in this case painting with a value of 0.4)

Reviewer: ```suggestion
In high deformation areas, such as around the mouth or under the eyes, add medium to low values (in this case painting with a value of 0.4).
```

Painting lower Attraction Force weights in meatier areas of the face, such as under the neck or in the cheecks to show more dynamics in these regions. In this case a value of 0.15 will be applied.

<figure>
  <img src="images/setup_simshape_4.png"> 
  <figcaption>Figure 29: Attraction Force weights for high dynamics areas.</figcaption>
</figure>

The lowest values (0.1 in this case) will be applied to the area under the jaw where dynamics will appear the most. 

<figure>
  <img src="images/setup_simshape_5.png"> 
  <figcaption>Figure 30: Attraction Force weights for highest dynamics areas.</figcaption>
</figure>

After painting similar weights to the ones displayed and pressing playback to check the animation,  realistic dynamics should be simulated in the face. Many more paintable weights to better customize and tweak face dynamics are avaliable and fully explained in the documentation for [AdnSimshape](simshape.md)

### 3. Adding muscle activations

To further have a realistic depiction of facial dynamics, facial muscle activations can be simulated. The AdnSimshape deformer has two methods of handling muscle activations:

 - Adonis Muscle Patches files.
 - Edge Evaluator Node.

> For this case the Edge Evaluator Node (Information on Adonis Muscle Patches files can be foud in the documentation for [AdnSimshape](simshape.md#muscle-activations)) can be used. To create this node, select the rest mesh, then the deformation mesh, and then go to AdonisFX Menu > Nodes > *Edge Evaluator*.

Reviewer: ```suggestion
For this case the Edge Evaluator Node can be used (information on Adonis Muscle Patches files can be foud in the documentation for [AdnSimshape](simshape.md#muscle-activations)). To create this node, select the rest mesh, then the deformation mesh, and then go to AdonisFX Menu > Nodes > *Edge Evaluator*.
```

Reviewer: Done.

Once the node is created it will appear in the Node Editor. Connect it to the AdnSimshape deformer. To do this go to AdonisFX Menu > Simshape (edit group) > *Connect Activations Plug*. 
<figure>
  <img src="images/setup_simshape_6.png"> 
  <figcaption>Figure 31: Connecting Edge Evaluator Node to AdnSimshape Deformer.</figcaption>
</figure>

> In the attribute editor of the AdnSimshape deformer, under the "Muscles Activation" section, the "Plug Values" will be enabled as a new valid "Activation Mode" option. To better visualize activations, press the ![AdnMuscle](images/adn_simshape_debugger.png) shelf button or go to AdonisFX Menu > Simshape (edit group) > *Activations Debugger*.

Reviewer: ```suggestion
In the attribute editor of the AdnSimshape deformer, under the *Muscles Activation* section, the *Plug Values* will be enabled as a new valid *Activation Mode* option. To better visualize activations, press the ![AdnMuscle](images/adn_simshape_debugger.png) shelf button or go to AdonisFX Menu > Simshape (edit group) > *Activations Debugger*.
```

Reviewer: Done.

# A Simple Setup

This page is dedicated to explain, step by step, the process of creating and setting every Adonis deformer in Maya. The scenarios set up here will be basic cases with the minimum configuration to start observing expected results in each case.

## AdnSkin simple setup

> To create a basic scenario using the AdnSkin deformer we will start with a scene with the following elements:

Reviewer 2: Please avoid the use of we wil and rewrite it in a more non-direct way. For example here:
```suggestion
To create a basic scenario using the AdnSkin deformer start with a scene with the following elements:
```

Please apply this to the whole document to make it more impersonal

Reviewer 2: Done

  - A target mesh with animation.
  - A simulated mesh (without animation).

The AdnSkin deformer will be applied to the simulated mesh.

<figure>
  <img src="images/setup_skin_0.png"> 
  <figcaption>Figure 1: Basic set up for skin simulations.</figcaption>
</figure>

### 1. Creating the deformer

> To create the deformer we will select the target mesh and then the skin mesh (simulated mesh). Then we will press the ![adnMuscle](images/adn_skin.png) shelf button or go to *AdonisFX menu > Skin*.

Reviewer 2: Following the suggestions from skin we are only using ** for the button. For example here it would be 
```suggestion
To create the deformer we will select the target mesh and then the skin mesh (simulated mesh). Then we will press the ![adnMuscle](images/adn_skin.png) shelf button or go to AdonisFX menu > *Skin*.
```

Please review this for the whole document

> To create the deformer we will select the target mesh and then the skin mesh (simulated mesh). Then we will press the ![adnMuscle](images/adn_skin.png) shelf button or go to *AdonisFX menu > Skin*.

Reviewer 2: ```suggestion
To create the deformer we will select the target mesh and then the skin mesh (simulated mesh). Then we will press the ![AdnMuscle](images/adn_skin.png) shelf button or go to *AdonisFX menu > Skin*.
```

Reviewer 2: Done

If we want to create the AdnSkin deformer with some initial customization we can double click the shelf button or press the option box in the menu item. This will display a popup window that will allow us to do some initial customization, as well as creating the deformer with a custom name. Once everything is in order press the "Create" button and the deformer will get created.

<figure>
  <img src="images/setup_skin_1.png"> 
  <figcaption>Figure 2: AdnSkin deformer creation scenario.</figcaption>
</figure>

### 2. Painting weights

> Now that we have the skin deformer properly created we are now able to apints its weights to correctly set up the deformer properties so it can behave as expected. To do so we must select the simulated mesh and press the ![paint tool](images/adn_paint_tool.png) shelf button or go to *AdonisFX menu > Paint Tool*.

Reviewer 2: ```suggestion
Now that we have the skin deformer properly created we are now able to paint its weights to correctly set up the deformer properties so it can behave as expected. To do so we must select the simulated mesh and press the ![paint tool](images/adn_paint_tool.png) shelf button or go to *AdonisFX menu > Paint Tool*.
```

Reviewer 2: Done

> We will start by painting **Hard Constraints**, which we will only apply to the edges of the simulated skin, so it properly attaches to the reference mesh.  

Reviewer 2: If I'm not mistaken now we willñ be using italics fro attributes. Please apply it to the whole document

Reviewer 2: Done

<figure>
  <img src="images/setup_skin_2.png"> 
  <figcaption>Figure 3: Hard Constraints weights paint.</figcaption>
</figure>

> Once we have that covered will then paint **Soft Constraints**, selecting it from the attribute enumerator. We will flood this weight to a low value of 0.2 to have a uniform distribution of soft constraints, so all the skin will properly follow the reference mesh while still simulating its own dynamics.

Reviewer 2: ```suggestion
Once we have that covered paint **Soft Constraints**, selecting it from the attribute enumerator. We will flood this weight to a low value of 0.2 to have a uniform distribution of soft constraints, so all the skin will properly follow the reference mesh while still simulating its own dynamics.
```

Reviewer 2: Done

<figure>
  <img src="images/setup_muscle_0.png"> 

>   <figcaption>Figure 6: Basic set up for biceps simulations.</figcaption>

Reviewer 2: ```suggestion
  <figcaption>Figure 6: Basic setup for biceps simulations.</figcaption>
```

Reviewer 2: Done

<figure>
  <img src="images/setup_muscle_8.png"> 
  <figcaption>Figure 15: Connection Editor tool.</figcaption>
</figure>

Now we will see that when the elbow gets flexed (and therefore the angle from the locator gets smaller) the muscle activation will get higher, simulating a much more realistic scenario.

> AdonisFX provides an endless array of customization and tweaking to fine tune the muscle dynamics by making use of these tools and the many additional parameters of the AdnMuscle deformer, that you can consult in the documentation for [AdnMuscle](muscle.md).

Reviewer 2: We should be careful with this. We are in docs not marketing. Please rephrase this without saying things like "endless array of customization and tweaking"

Reviewer 2: With that being said we might want to expose an example for setting up slide on segment constraints

Reviewer: The rephrasing was already addressed.
